### PR TITLE
enh(core): display request_id in ModelError

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

adds `request_id` to the `ModelError` struct and populate it when possible (i.e, when we have a response and the header is present in the response)

## Risk

N/A

## Deploy Plan

N/A

### CODE REVIEW

File: core/src/providers/llm.rs [hunk 1]
```diff
@@ -493,12 +495,26 @@ impl LLMChatRequest {
...
-                self.provider_id.to_string(),
-                self.model_id,
-                e.to_string(),
-            )),
+            Err(e) => {
+                let string_error = e.to_string();
+                if let Ok(e) = e.downcast::<ModelError>() {
+                    //  Log the request_id if it exists.
+                    if let Some(request_id) = e.request_id {
+                        info!(
```
Comment (spolu) [inline core/src/providers/llm.rs:503]: `error!` ? Even if ~ similar for now might be better in the future?

**Review by spolu — APPROVED** [review]

File: core/src/providers/llm.rs (hunk 1 above)
Comment (fontanierh) [reply core/src/providers/llm.rs]: fixed. And improved the message as well

Comment (fontanierh) [reply core/src/providers/llm.rs]: And also added to the execute of regular llm request

File: core/src/providers/llm.rs [hunk 2]
```diff
@@ -293,12 +295,26 @@ impl LLMRequest {
...
+            Err(e) => {
+                let string_error = e.to_string();
+                if let Ok(e) = e.downcast::<ModelError>() {
+                    //  Log the request_id if it exists.
+                    if let Some(request_id) = e.request_id {
+                        error!(
+                            provider_id = self.provider_id.to_string(),
+                            model_id = self.model_id,
+                            request_id = request_id,
+                            "Too many retries: error=ModelError"
```
Comment (spolu) [inline core/src/providers/llm.rs:307]: Why Too many retries?

Comment (fontanierh) [reply core/src/providers/llm.rs:307]: We end up here if `with_retryable_back_off` failed right ?

Comment (fontanierh) [reply core/src/providers/llm.rs:307]: Happy to change it back to "Failed to query" though

Comment (spolu) [reply core/src/providers/llm.rs:307]: Generally yes that string is already injected in the error message. (see provider.rs)

Which means you'll likely won't have a ModelError here ever also.

Which means you want that log in `with_retryable_back_off` IMHO ?

Comment (fontanierh) [reply core/src/providers/llm.rs:307]: Indeed 👍

**Pull request merged**